modules/l4socks/socks4_matcher.go

```diff
@@ -0,0 +1,122 @@
+package l4socks
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/caddyserver/caddy/v2"
+	"github.com/mholt/caddy-l4/layer4"
+	"io"
+	"net"
+	"strings"
+)
+
+func init() {
+	caddy.RegisterModule(&Socks4Matcher{})
+}
+
+// Socks4Matcher matches SOCKSv4 connections according to https://www.openssh.com/txt/socks4.protocol.
+// Since the SOCKSv4 header is very short it could produce a lot of false positives.
+// To improve the matching use Commands, Ports and Networks to specify to which destinations you expect clients to connect to.
+// By default CONNECT & BIND commands are matched with any destination ip and port.
+type Socks4Matcher struct {
+	// Only match on these commands. Default: ["CONNECT", "BIND"]
+	Commands []string `json:"commands,omitempty"`
+	// Only match on requests to one of these destination networks (IP or CIDR). Default: all networks.
+	Networks []string `json:"networks,omitempty"`
+	// Only match on requests to one of these destination ports. Default: all ports.
+	Ports []uint16 `json:"ports,omitempty"`
+
+	commands []uint8
+	cidrs    []*net.IPNet
+}
+
+func (m *Socks4Matcher) CaddyModule() caddy.ModuleInfo {
```

Comment:
Can you explain me why this should be changed? Is there a caddy specific reason? Because my IDE is warning me about mixed value and pointer receivers and points me to [Go FAQ](https://go.dev/doc/faq#methods_on_values_or_pointers)
> Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used.

I will also change it for the socks5 matcher/handler which do the same.

Reply:
:man_shrugging: I've never seen an error like that. The matcher value (m) is not ever used in the method anyway, so it shouldn't matter.

Reply:
It's only a weak warning 😄 So I guess it's just a matter of preference.